grpc/errors.go

```diff
@@ -63,59 +70,86 @@ func unwrapError(err error, md metadata.MD) error {
 		return nil
 	}
 
-	unwrappedErr := status.Convert(err).Message()
-
 	errTypeStrs, ok := md["errortype"]
 	if !ok {
 		return err
 	}
+
+	inErrMsg := status.Convert(err).Message()
 	if len(errTypeStrs) != 1 {
 		return berrors.InternalServerError(
-			"multiple errorType metadata, wrapped error %q",
-			unwrappedErr,
+			"multiple 'errortype' metadata, wrapped error %q",
+			inErrMsg,
 		)
 	}
 
-	errType, decErr := strconv.Atoi(errTypeStrs[0])
+	inErrType, decErr := strconv.Atoi(errTypeStrs[0])
 	if decErr != nil {
 		return berrors.InternalServerError(
 			"failed to decode error type, decoding error %q, wrapped error %q",
 			decErr,
-			unwrappedErr,
+			inErrMsg,
 		)
 	}
-	outErr := berrors.New(berrors.ErrorType(errType), unwrappedErr)
-
-	subErrsJSON, ok := md["suberrors"]
-	if !ok {
-		return outErr
-	}
-	if len(subErrsJSON) != 1 {
-		return berrors.InternalServerError(
-			"multiple suberrors metadata, wrapped error %q",
-			unwrappedErr,
+	inErr := berrors.New(berrors.ErrorType(inErrType), inErrMsg)
+	var inBoulderErr *berrors.BoulderError
+	if !errors.As(inErr, &inBoulderErr) {
+		return fmt.Errorf(
+			"expected type of unwrappedErr to be %T got %T: %q",
+			inBoulderErr,
+			inErr,
+			inErr.Error(),
 		)
 	}
 
-	var suberrs []berrors.SubBoulderError
-	err2 := json.Unmarshal([]byte(subErrsJSON[0]), &suberrs)
-	if err2 != nil {
-		return berrors.InternalServerError(
-			"error unmarshaling suberrs JSON %q, wrapped error %q",
-			subErrsJSON[0],
-			unwrappedErr,
-		)
+	var decodedErrWithSubErrs *berrors.BoulderError
+	subErrorsVal, ok := md["suberrors"]
+	if ok {
+		if len(subErrorsVal) != 1 {
+			return berrors.InternalServerError(
+				"multiple 'suberrors' in metadata, wrapped error %q",
+				inErrMsg,
+			)
+		}
+
+		var subErrs []berrors.SubBoulderError
+		unmarshalErr := json.Unmarshal([]byte(subErrorsVal[0]), &subErrs)
+		if unmarshalErr != nil {
+			return berrors.InternalServerError(
+				"JSON unmarshaling 'suberrors' %q, wrapped error %q: %s",
+				subErrorsVal[0],
+				inErrMsg,
+				unmarshalErr,
```

Comment:
Not sure if we were omitting this error on purpose here. Let me know if I should return it to it's original state.

Reply:
I don't think we were omitting it on purpose. Nice catch!